mongo/src/main/scala/it/agilelab/darwin/connector/mongo/MongoConnector.scala

```diff
@@ -1,19 +1,165 @@
+  require(config.hasPath(ConfigurationKeys.TABLE))
+
+  log.debug("Creating default MongoConfiguration")
+  val mongoConfig: MongoConfig = MongoConfig(
+    config.getString(ConfigurationKeys.USER),
+    config.getString(ConfigurationKeys.PASSWORD),
+    config.getStringList(ConfigurationKeys.HOST).asScala.toList,
+    config.getString(ConfigurationKeys.DATABASE),
+    config.getString(ConfigurationKeys.TABLE)
+  )
+  log.debug("Created default MongoConfiguration")
+
+  log.debug(s"Creating connections")
+  val mongoClient: MongoClient = createConnection(mongoConfig)
+  log.debug(s"Connection created")
+
+  //  In MongoDB tables are called collections.
+  //  Schema:
+  //    id:        idSchema         (long)
+  //    schema:    schema           (string)
+  //    name:      schema.name      (string)
+  //    namespace: schema.namespace (string)
+
+  /**
+   * return the MongoClient
+   * @param mongoConf : config with keys defined by ConfigurationKeys object
+   * @return MongoClient
+   */
+  private def createConnection(mongoConf: MongoConfig): MongoClient = {
+
+    val credential: MongoCredential = createCredential(mongoConf.user, mongoConf.db, mongoConf.password.toCharArray)
+
+    val hosts: List[ServerAddress] = mongoConf.hosts.map(host => new ServerAddress(host))
+
+    val settings: MongoClientSettings = MongoClientSettings.builder()
+      .credential(credential)
+      .applyToClusterSettings(builder => builder.hosts(hosts.asJava))
+      //.streamFactoryFactory(NettyStreamFactoryFactory())
+      .build()
+
+    MongoClient(settings)
+
+  }
+
+  private def parser: Parser = new Parser()
+
+  override def fullLoad(): Seq[(Long, Schema)] = {
+    log.debug(s"loading all schemas from table ${mongoConfig.table}")
+    val collectionNames =
+      mongoClient
+        .getDatabase(mongoConfig.db)
+        .listCollectionNames()
+
+    val schemas: Seq[(Long, Schema)] =
+      Await.result(
+        collectionNames.flatMap(collectionName =>
+          mongoClient.getDatabase(mongoConfig.db).getCollection(collectionName).find().map(document => {
+            val key = document.filterKeys(k => k == "id").head._2.asInt64().getValue
+            val schema = parser.parse(document.filterKeys(k => k == "schema").head._2.asString().getValue)
+            key -> schema
+          })
+        ).toFuture()
+        ,
+        Duration.Undefined
+      )
+
+    log.debug(s"${schemas.size} loaded from MongoDB")
+    schemas
+  }
+
+  override def insert(schemas: Seq[(Long, Schema)]): Unit = {
+
+    schemas.foreach{
+      case (id, schema) => {
+        val document = new BsonDocument
+        document.put("id", bson.BsonInt64(id))
+        document.put("schema", bson.BsonString(schema.toString))
+        document.put("name", bson.BsonString(schema.getName))
+        document.put("namespace", bson.BsonString(schema.getNamespace))
+
+        mongoClient.getDatabase(mongoConfig.db).getCollection(mongoConfig.table).insertOne(document)
```

Comment:
what's the semantic of insert? Does it fail if the key already exists? If so, we should use a different operation